server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderator.java

```diff
@@ -1516,15 +1519,19 @@ public Void apply(@Nullable Object input)
               }
             }
 
-            unannounceAllVersionsOfSegment(sink.getSegment(), sink);
+            final Set<SegmentIdWithShardSpec> allVersionIds = unannounceAllVersionsOfSegment(sink.getSegment(), sink);
 
             Runnable removeRunnable = () -> {
-              droppingSinks.remove(baseIdentifier);
+              droppingSinks.remove(identifier);
               sinkTimeline.remove(
                   sink.getInterval(),
                   sink.getVersion(),
-                  baseIdentifier.getShardSpec().createChunk(sink)
+                  identifier.getShardSpec().createChunk(sink)
               );
+              for (SegmentIdWithShardSpec id : allVersionIds) {
+                // Update query mapping with SinkQuerySegmentWalker
+                ((SinkQuerySegmentWalker) texasRanger).unregisterUpgradedPendingSegment(id);
```

Comment:
If `allVersionIds` does not contain the `baseIdentifier`, `unregister` might need to be called separately for it.

Reply:
It contains the baseIdentifier